### PR TITLE
Fix BC.PolicyUpdate double sending

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Fixes https://github.com/smartdevicelink/sdl_core/issues/3138

This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Testing Plan
http://172.26.147.12/view/Issue_3138/

### Summary
Sometimes SDL sends BasicCommunication.PolicyUpdate twice.
I believe that the main problem is that when SDL receives the second PTU trigger, in the PolicyManagerImpl::StartPTExchange() ->  update_status_manager_.IsUpdatePending() returned **false** because the state is still "UPDATE_NEEDED". 

On the latest release - SDL should change the status to "UPDATING" right after sending the BasicCommunication.PolicyUpdate to the HMI. But in some cases, SDL can process the second PTU trigger before changing the status to the "UPDATING".

Taking into account all the above I have created another one internal status "UPDATE_PENDING" and use this status for cases when SDL knows that update is required and until the snapshot will be sent to the HMI. 
So SDL will change the status to "UPDATE_PENDING" before calling RequestPTUpdate(); in PolicyManagerImpl::StartPTExchange(); -> IsUpdatePending() for "UPDATE_PENDING" will return **true**; ->  After sending the first BC.PolicyUpdate the status will be changed to "UPDATING" -> SDL won't request a PolicyUpdate twice

### CODE REVIEW

File: src/components/policy/policy_external/include/policy/update_status_manager.h [hunk 1]
```diff
@@ -149,6 +149,8 @@ class UpdateStatusManager {
    */
   void ScheduleUpdate();
 
+  void PendingUpdate();
```
Comment (mkorniichuk) [inline src/components/policy/policy_external/include/policy/update_status_manager.h]: @LitvinenkoIra how about doxygen?

File: src/components/policy/policy_external/src/update_status_manager.cc [hunk 2]
```diff
@@ -164,9 +165,15 @@ bool UpdateStatusManager::IsUpdatePending() const {
...
+  LOG4CXX_AUTO_TRACE(logger_);
   ProcessEvent(kScheduleUpdate);
 }
 
+void UpdateStatusManager::PendingUpdate() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  ProcessEvent(kPendingUpdate);
+}
+
 void UpdateStatusManager::ScheduleManualUpdate() {
```
Comment (mkorniichuk) [inline src/components/policy/policy_external/src/update_status_manager.cc]: @LitvinenkoIra good place for **`LOG4CXX_AUTO_TRACE`**

**Comment by JackLivio** [comment]
@LitvinenkoIra Was the original issue only seen in external policies?

File: src/components/policy/policy_external/include/policy/status.h [hunk 3]
```diff
@@ -161,6 +162,37 @@ class UpdateNeededStatus : public Status {
   bool IsUpdateRequired() const OVERRIDE;
 };
 
+/**
+ * @brief The UpdatePendingStatus class represents 'update pending' status
```
Comment (JackLivio) [inline src/components/policy/policy_external/include/policy/status.h]: I think the description of the class should be more functional. The reasoning you gave in the PR description was really good. Maybe use some of that description here?

> The UpdatePendingStatus class represents cases when SDL knows that update is required and but before the snapshot is sent to the HMI.

**Comment by LitvinenkoIra** [comment]
@JackLivio Yes, the original issue was only seen in external policy in case SDL receives two PTU triggers one by one

File: src/components/policy/policy_external/src/update_status_manager.cc (hunk 2 above)
Comment (LitvinenkoIra) [reply src/components/policy/policy_external/src/update_status_manager.cc]: @mkorniichuk Done in 7d5222c6ca6e483113b51eae3332d9e94ec5444a

File: src/components/policy/policy_external/include/policy/status.h (hunk 3 above)
Comment (LitvinenkoIra) [reply src/components/policy/policy_external/include/policy/status.h]: @JackLivio Done in 7d5222c6ca6e483113b51eae3332d9e94ec5444a

**Comment by dboltovskyi** [comment]
@theresalech Please notice this PR is ready for PM review

**Pull request merged**